tests/cases/unittests/transpile.ts

```diff
@@ -274,5 +274,62 @@ var x = 0;`,
         it("Supports backslashes in file name", () => {
             test("var x", { expectedOutput: "var x;\r\n", options: { fileName: "a\\b.ts" }});
         });
+        
+        it("transpile file as 'tsx' if 'jsx' is specified", () => {
+            let input = `import * as React from 'react';\r\n` +

```

Comment:
This test will break if we change almost anything about how `React` emit works; a shorter example (just using `var x = <div />;` for example) is much less likely to create a maintenance hassle in that regard.